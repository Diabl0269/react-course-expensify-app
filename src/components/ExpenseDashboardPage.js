import React from "react";
import ExpesneList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpesneList />
  </div>
);

export default ExpenseDashboardPage;
