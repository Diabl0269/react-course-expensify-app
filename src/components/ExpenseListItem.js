import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = ({ id, description, amount, createdAt, note }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>Description: {description}</h3>
    </Link>
    Amount: {numeral(amount / 100).format("$0,0.00")}, Created at:{" "}
    {moment(createdAt).format("MMM Do, YYYY")}, Note:
    {note && ` ${note}`}
  </div>
);

export default ExpenseListItem;
