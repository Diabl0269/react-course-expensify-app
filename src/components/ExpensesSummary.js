import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import expensesTotal from '../selectors/expenses-total';
import numeral from "numeral";


export class ExpensesSummary extends React.Component {
  render() {
  return <div>
      Number of Expenses: {this.props.expenses.length} 
      <br />
      Total expenses: {numeral(expensesTotal(this.props.expenses) / 100).format("$0,0.00") }
      </div>;
  };
};

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
