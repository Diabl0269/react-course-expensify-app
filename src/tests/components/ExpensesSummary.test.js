import React from "react";
import { shallow } from "enzyme";
import {ExpensesSummary} from "../../components/ExpensesSummary";
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with no expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpensesSummary with expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});