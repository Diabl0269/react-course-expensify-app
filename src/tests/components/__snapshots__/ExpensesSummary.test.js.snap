// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expenses correctly 1`] = `
<div>
  Number of Expenses: 
  3
  <br />
  Total expenses: 
  $1,141.95
</div>
`;

exports[`should render ExpensesSummary with no expenses correctly 1`] = `
<div>
  Number of Expenses: 
  0
  <br />
  Total expenses: 
  $0.00
</div>
`;
